clang/include/clang/Sema/Sema.h

```diff
@@ -12580,10 +12580,15 @@ class Sema final {
 
   static const Type* getBaseType(const Type* pt)
   {
-    while (pt->isAnyPointerType())
```

Comment:
I am suspicious about looping over pointers. What is the rationale for that?

Reply:
Looping over pointers to get the basic type was introduced in 44515b5c362d733e2dce69fbb7c7e59dfae9e1b9 and now it's used more generally to find the basic type kind (genericjs / asmjs).